lib/launchers/ocm.rb

```diff
@@ -0,0 +1,238 @@
+    def generate_json(name)
+      json_data = {
+        "name" => name,
+        "managed" => true,
+        "multi_az" => false,
+        "byoc" => false
+      }
+
+      if @multi_az
+        json_data.merge!({"multi_az" => @multi_az})
+      end
+
+      if @region
+        json_data.merge!({"region" => {"id" => @region}})
+      end
+
+      if @version
+        json_data.merge!({"version" => {"id" => "openshift-v#{@version}"}})
+      end
+
+      if @nodes
+        json_data.merge!({"nodes" => {"compute" => @nodes.to_i}})
+      end
+
+      if @lifespan
+        expiration = Time.now + to_seconds(@lifespan)
+        json_data.merge!({"expiration_timestamp" => expiration.strftime("%Y-%m-%dT%H:%M:%SZ")})
+      end
+
+      if @aws_account_id && @aws_access_key && @aws_secret_key
+        json_data.merge!({"aws" => {"account_id":@aws_account_id, "access_key_id":@aws_access_key, "secret_access_key":@aws_secret_key}})
+        json_data.merge!({"byoc" => true})
+      end
+
+      return json_data.to_json
+    end
+
+    # download the script 'osd-provision.sh' which takes care of the OSD installation/uninstallation
+    def download_osd_script
+      osd_repo_uri = ENV['GIT_OSD_URI'] || 'https://gitlab.cee.redhat.com/mk-bin-packing/mk-performance-tests.git'
+      osd_repo_dir = File.join(Dir.tmpdir, 'osd_repo')
+      FileUtils.rm_rf(osd_repo_dir)
+      git = BushSlicer::Git.new(uri: osd_repo_uri, dir: osd_repo_dir)
+      git.clone
+      osd_script = File.join(osd_repo_dir, 'scripts', 'osd-provision.sh')
+      if !File.exists?(osd_script)
+        raise "Cannot find #{osd_script}"
+      end
+      return osd_script
+    end
+
+    def shell(cmd, output = nil)
+      if output
+        res = Host.localhost.exec(cmd, single: true, stderr: :stdout, stdout: output, timeout: 3600)
+      else
+        res = Host.localhost.exec(cmd, single: true, timeout: 3600)
+      end
+      if res[:success]
+        return res[:response]
+      else
+        raise "Error when executing '#{cmd}'. Response: #{res[:response]}"
+      end
+    end
+
+    def get_last_line(text)
+      last_line = nil
+      text.each_line { |line| last_line = line }
+      return last_line
+    end
+
+    # generate OCP information
+    def generate_ocp_info(api_url, json_creds)
+      api_regex = /https?:\/\/api\.([\S]+):[\d]*/
+      if api_url.match(api_regex)
+        domain = api_url.scan(api_regex).first.first
+      else
+        raise "Given api_url '#{api_url}' doesn't match '#{api_regex}'"
+      end
+      credentials = JSON.parse(json_creds)
+      ocp_info = {
+        "ocp_domain" => domain,
+        "ocp_api_url" => "https://api.#{domain}:6443",
+        "ocp_console_url" => "https://console-openshift-console.apps.#{domain}",
+        "user" => credentials["user"],
+        "password" => credentials["password"]
+      }
+      return ocp_info
+    end
+
+    # create OSD cluster
+    def create_osd(name)
```

Comment:
probably makes sense to check if cluster exists already and/or keep status of that within the object
I mean before attempting an installation.

Reply:
If an error occurs then it is returned by the `ocm-cli` tool. So, no need to extract some checks from that tool.